pkg/daemon/pinned_image_set.go

```diff
@@ -0,0 +1,1273 @@
+	"fmt"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/docker/reference"
+	ign3types "github.com/coreos/ignition/v2/config/v3_4/types"
+	"github.com/golang/groupcache/lru"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformersv1 "k8s.io/client-go/informers/core/v1"
+	corev1lister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
+
+	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	mcfgv1alpha1 "github.com/openshift/api/machineconfiguration/v1alpha1"
+	machineconfigurationalphav1 "github.com/openshift/client-go/machineconfiguration/applyconfigurations/machineconfiguration/v1alpha1"
+	mcfgclientset "github.com/openshift/client-go/machineconfiguration/clientset/versioned"
+	mcfginformersv1 "github.com/openshift/client-go/machineconfiguration/informers/externalversions/machineconfiguration/v1"
+	mcfginformersv1alpha1 "github.com/openshift/client-go/machineconfiguration/informers/externalversions/machineconfiguration/v1alpha1"
+	mcfglistersv1 "github.com/openshift/client-go/machineconfiguration/listers/machineconfiguration/v1"
+	mcfglistersv1alpha1 "github.com/openshift/client-go/machineconfiguration/listers/machineconfiguration/v1alpha1"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/daemon/constants"
+	"github.com/openshift/machine-config-operator/pkg/daemon/cri"
+	"github.com/openshift/machine-config-operator/pkg/helpers"
+	"github.com/openshift/machine-config-operator/pkg/upgrademonitor"
+)
+
+const (
+	// cri gRPC connection parameters taken from kubelet
+	criPrefetchInterval             = 30 * time.Second
+	defaultPrefetchWorkers          = 5
+	defaultControlPlaneWorkers      = 1
+	defaultPrefetchThrottleDuration = 1 * time.Second
+
+	crioPinnedImagesDropInFilePath = "/etc/crio/crio.conf.d/50-pinned-images"
+
+	// backoff configuration
+	maxRetries    = 5
+	retryDuration = 1 * time.Second
+	retryFactor   = 2.0
+	retryCap      = 10 * time.Second
+
+	// controller configuration
+	maxRetriesController = 15
+)
+
+var (
+	errInsufficientStorage       = errors.New("storage available is less than minimum required")
+	errFailedToPullImage         = errors.New("failed to pull image")
+	errNotFound                  = errors.New("not found")
+	errNoAuthForImage            = errors.New("no auth found for image")
+	errImageSetGenerationUpdated = errors.New("image set generation updated")
```

Comment:
Is this error actually being used somewhere?

Reply:
will remove I just deleted the code that used it.